### DISCUSSION TITLE
from racoon to strongswan

### DISCUSSION THREAD

**Discussion by havre-gryn** [strongswan/strongswan#2087] [Q&A] [answered]
I'm making the switch from racoon to strongswan/charon, 
and have a racoon config that I am trying to replicate in swanctl.conf
The connection is to a Cisco ASA with IKEv1, tunnel group name and preshared key

racoon config that is working today:
```
remote 192.168.16.174 {
	exchange_mode aggressive;
	script "/etc/ipsec/p1_up_down.sh" phase1_up;
	script "/etc/ipsec/p1_up_down.sh" phase1_down;
	script "/etc/ipsec/p1_up_down.sh" phase1_dead;
	proposal {
	lifetime time 1800 seconds;
		encryption_algorithm aes 128;
		hash_algorithm sha1;
		authentication_method xauth_psk_client;
		dh_group 2;
	}
	generate_policy on;
	nat_traversal force;
	my_identifier keyid tag "tunnel_jal1";
	verify_identifier off;
	mode_cfg on;
	nonce_size 16;
	initial_contact on;
	proposal_check obey;
	xauth_login "jal1";
}
```
The racoon "psk.txt"
```
192.168.16.174	presharedkey
tunnel_jal1	presharedkey
jal1
```
Attempt at swanctl.conf (which is not working):

```
connections {
	 home {
		 version = 1
		 proposals = aes128-sha1-modp1024
		 aggressive = yes
		 remote_addrs = 192.168.16.174
		 vips = 0.0.0.0
		 encap = no

		 local-psk {
			round=1
		   auth = psk
		   id = "@#74756e6e656c5f6a616c31"
		 }
                remote-psk {
                        round=1
                         auth = psk
                         id = %any
                 }

#		 local {
#		   id = keyid:tunnel_jal1
#		 }

		 local-xauth {
			round=2
		   auth = xauth
		   xauth_id = jal1
		 }
                 remote-xauth {
                        round=2
                   auth = xauth
                   id = %any
                 }

		 children {
			 sweet_child_o_mine {
				 remote_ts  = 192.168.1.0/24
				 start_action = trap
				 esp_proposals = aes128-sha1-modp1024
			 }
		 }
	 }
}
secrets {
	 ike-group {
		 secret = presharedkey
		   id = "@#74756e6e656c5f6a616c31"
	 }
	 xauth {
		 id = jal1
		 secret = crap
	 }
}

```

Issue is most likely with the tunnel group and xauth combination?
The Cisco does not complain in its' syslog, Cisco responds to initial packet but then the Phase1 stalls and times out. 

Roadwarrior(strongswan) client : 192.168.16.124 
Cisco ASA : 192.168.16.174

13:05:59.910003 IP 192.168.16.124.500 > 192.168.16.174.500: isakmp: phase 1 I agg
13:05:59.912819 IP 192.168.16.174.500 > 192.168.16.124.500: isakmp: phase 1 R agg
13:06:07.910335 IP 192.168.16.174.500 > 192.168.16.124.500: isakmp: phase 1 R agg
13:06:15.910484 IP 192.168.16.174.500 > 192.168.16.124.500: isakmp: phase 1 R agg
13:06:23.910481 IP 192.168.16.174.500 > 192.168.16.124.500: isakmp: phase 1 R agg

I can provide additional info and logs if needed.

How do I create a strongswan config that replicates the racoon config?

**Comment by havre-gryn** [comment]
uploaded log from strongswan
[ike_log.txt](https://github.com/strongswan/strongswan/files/14112185/ike_log.txt)

**Comment by Thermi** [comment, marked as answer]
Try removing remote xauth and specifying local id as keyid"tunnel_jal1 for psk phase.

**Reply by havre-gryn** [reply]
Thanks a lot Thermi !

Yes, removing remote xauth made phase1 progress further.
The keyid was okay, specified with the ascii codes, but your suggestion "keyid:" makes it more readable.

Then xauth failed. 
I have the Cisco set up with empty password (only a username configured)
Strongswan configuration cannot accept an empty password [secret = ""]  or [secret = ]
Whereas racoon does accept configuration of an empty password.

Changing the password on the Cisco to something meaningful, and adding that in the strongswan config, made it work all the way
Great!

**Reply by Thermi** [reply]
Those are great news. Congratulations!

**Comment by havre-gryn** [comment]
The swanctl.conf  that is now working against the Cisco:

```
root@yop~ cat /var/run/vpn/ipsec/swanctl.conf 
connections {
	 home {
		 version = 1
		 proposals = aes128-sha1-modp1024
		 aggressive = yes
		 remote_addrs = 192.168.16.174
		 vips = 0.0.0.0
		 encap = yes

#		 local-psk {
#		   auth = psk
#		   id = "@#74756e6e656c5f6a616c31"
#		 }

		 local-psk {
                   auth = psk
		   id = keyid:tunnel_jal1
		 }

		 local-xauth {
		   auth = xauth
		   xauth_id = jal1
		 }
		 remote-psk {
			 auth = psk
			 id = %any
		 }
		 children {
			 sweet_child_o_mine {
				 remote_ts  = 192.168.1.0/24
				 start_action = trap
				 esp_proposals = aes128-md5-modp768
			 }
		 }
	 }
}
secrets {
	 ike-group {
		 secret = presharedkey
#		   id = "@#74756e6e656c5f6a616c31"
		   id = keyid:tunnel_jal1
	 }
	 xauth {
		 id = jal1
		 secret = pass
	 }
}

```
